mythril_core/src/acpi/rsdt.rs

```diff
@@ -133,6 +137,17 @@ impl<'a> RSDT<'a> {
             })
             .unwrap_or(Err(Error::NotFound))
     }
+
+    /// Returns true if self is a reference to an XSDT
+    /// so that entries can be treated as 64-bit addresses.
+    pub fn is_xsdt(&self) -> bool {
```

Comment:
Rather than relying on comparing the signature string, I think this is a good spot for an enum. Maybe something like:

```
enum RSDT<'a> {
    RSDT(SDT<'a>),
    XSDT(SDT<'a>),
}
```

Reply:
Thanks for the review! always a good learning experience.

`acpi -2` may indeed be a good idea. I'll add it.

enum would definitely work better, I was just having a hard time thinking of how to implement it. Appreciate your nudge in the right direction

Reply:
Done and done.

Let me know what you think of my enum implementations, please